bedwars-plugin/src/main/java/com/tomkeuper/bedwars/commands/bedwars/subcmds/regular/CmdLeave.java

```diff
@@ -53,12 +59,37 @@ public boolean execute(String[] args, CommandSender s) {
         if (s instanceof ConsoleCommandSender) return false;
         Player p = (Player) s;
 
-        if (cancel(p.getUniqueId())) return true;
-        update(p.getUniqueId());
-        IArena a = Arena.getArenaByPlayer(p);
-
-        Misc.moveToLobbyOrKick(p, a, a != null && a.isSpectator(p.getUniqueId()));
-        return true;
+        if (p.getWorld().getName().equalsIgnoreCase(BedWars.getLobbyWorld())) {
+            if (cancel(p.getUniqueId())) return true;
+            update(p.getUniqueId());
+            IArena a = Arena.getArenaByPlayer(p);
+            Misc.moveToLobbyOrKick(p, a, a != null && a.isSpectator(p.getUniqueId()));
+            return true;
+        } else {
+            if (cancel(p.getUniqueId())) return true;
+            update(p.getUniqueId());
+            IArena a = Arena.getArenaByPlayer(p);
+            if (a == null) {
+                p.sendMessage(Language.getMsg(p, Messages.COMMAND_FORCESTART_NOT_IN_GAME));
+                return true;
+            }
+            PlayerGoods pg = PlayerGoods.getPlayerGoods(p);
+            if (pg != null && pg.getQuitTask() != null) {
+                Bukkit.getScheduler().cancelTask(pg.getQuitTask().getTaskId());
+                pg.setQuitTask(null);
+                p.sendMessage(Language.getMsg(p, Messages.LEAVE_CANCEL));
+                return true;
+            }
+            p.sendMessage(Language.getMsg(p, Messages.LEAVE_STARTED));
+            BukkitTask task = new BukkitRunnable() {
+                public void run() {
+                    Misc.moveToLobbyOrKick(p, a, a != null && a.isSpectator(p.getUniqueId()));
+                    pg.setQuitTask(null);
+                }
+            }.runTaskLater(BedWars.plugin, 60L);
```

Comment:
maybe make the leave delay configurable and add a check, if 0, then dont send all the messages just do it the old way

Reply:
> maybe make the leave delay configurable and add a check, if 0, then dont send all the messages just do it the old way

Okay. I am currently making

Reply:
> maybe make the leave delay configurable and add a check, if 0, then dont send all the messages just do it the old way

done